OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -124,7 +124,7 @@ protected virtual bool CanAttack(Actor self, Target target)
 
 		public virtual void DoAttack(Actor self, Target target, IEnumerable<Armament> armaments = null)
 		{
-			if (armaments == null && !CanAttack(self, target))
+			if (!CanAttack(self, target))
```

Comment:
Ok, so the reason this was excluding the `armaments == null` case is that `CanAttack` (quite correctly, IMO) uses the proper `Armaments` list instead of the ones passed in to this method.

It looks like the `armaments` override was added as a way to try and reduce duplicated code in the `Attack` activity (which has its own method of pre-filtering armaments), but IMO this has done more harm in terms of confusing the API than it has in removing a couple of lines of code.

Can we remove the `armaments` override here and then duplicate the `IsPaused` and `CheckFire` calls directly inside `Attack` instead?

Reply:
Done.